notes/conds.md

```diff
@@ -0,0 +1,83 @@
+# delegation clause
+
+In 0.x verison of UCANs has an open-ended `nb` field that most users use to constraints delegated capabilities. Open ended nature makes interop very difficult as how to interpert those constraints are not covered by spec. In 1.0 we would like to define standard but extensible constarints language so that implementations can interop unless user space extensions are utilized. By allowing user space extensions of the constraint language we also allow user space to innovate at the cost of interop.
+
+## s-expression based language
+
+### Operators
+
+| Operator     | Function Signature                | Description                                | Example                                        |
+| ------------ | --------------------------------- | ------------------------------------------ | ---------------------------------------------- |
+| `/${string}` |`[] -> Any`                        | Selects value using JSON Pointer            | `["/foo/0"]`                                  |
+|              |`[] -> Any`                        | Selects whole value like "" in JSON Pointer | `[]`                                          |
+| `==`         |`[Exp<Any>, Exp<Any>] -> Bool`     | Equality                                    | `["==", ["/size"], 1]`                        |
+| `<`          |`[Exp<Num>, Exp<Num>] -> Bool`     | Less than                                   | `["<",  ["/size"], 1024]`                     |
+| `<=`         |`[Exp<Num>, Exp<Num>] -> Bool`     | Less than or equal                          | `["<=", ["/size"], 1024]`                     |
+| `>`          |`[Exp<Num>, Exp<Num>] -> Bool`     | Greater than                                | `[">", ["@", "/size"], 0]`                    |
+| `>=`         |`[Exp<Num>, Exp<Num>] -> Bool`     | Greater than or equal                       | `[">=", ["@", "/size"], 32]`                  |
+| `not`        |`[Expr<Bool>] -> Bool`             | Negation                                    | `["not", ["<=", ["/size"], 128]`              |
+| `some`       |`[Expr<Any[]>, Expr<Bool>] -> Bool`| Collection predicate                        | `["some", ["/to"], ["==", [], "hi@web.mail"]]`|
```

Comment:
We no longer need some/every, because we can use unification:

```js
["$args", "foo.bar", "?x],
["match", "?x", "*exmaple.com"],
["?x", "!=", []] // AKA "some"
```

Reply:
These table was for s-expression variant of the notation and not for the logic program one.

Reply:
Ah gotcha 👍

Reply:
Thinking about this more: checking always ends in unification. That means that we need an `==`, `some`, or `every` somewhere in every query under normal Datalog rules (i.e. we need to pick cardinality).

I think I'm comfortable implicitly having a `every`, but I think we should also have an explicit one for people to use. It is clearer with the explicit version, but it would be annoying to have to start every statement with `every ?x ?y`

Reply:
We also have pattern matching, so this also deviates from datalog purely as a selector:
```
 "policy": [
     ["$args", "foo.bar", "hello world"] 
  ]
```

Normal Datalog would just return `[]` here. We have an implicit hook at the top level that says "every query in the policy array must be populated"